src/Extensions/ExceptionReporter.php

```diff
@@ -174,22 +192,44 @@ protected function createStackFrame(array $entry): StackFrame
 
         $frame = new StackFrame($filename, $function, $lineno);
 
-        if ($filename !== "{no file}") {
-            $this->loadContext($frame, $filename, $lineno, 5);
+        if ($this->root_path && strpos($filename, $this->root_path) !== -1) {
+            $frame->abs_path = $filename;
+            $frame->filename = substr($filename, strlen($this->root_path));
+        }
 
-            if ($this->root_path && strpos($filename, $this->root_path) !== -1) {
-                $frame->abs_path = $filename;
-                $frame->filename = substr($filename, strlen($this->root_path));
+        if ($this->isBlacklisted($filename)) {
+            $frame->context_line = "### BLACKLISTED ###";
+        } else {
+            if ($filename !== self::NO_FILE) {
+                $this->loadContext($frame, $filename, $lineno, 5);
             }
-        }
 
-        if (isset($entry['args'])) {
-            $frame->vars = $this->extractVars($entry);
+            if (isset($entry['args'])) {
+                $frame->vars = $this->extractVars($entry);
```

Comment:
Maybe we can consider having our own var_name scrubber here.

Reply:
We could, but if Sentry is going to sanitize it anyhow, pre-storage, and the transport is over SSL, I'm not that concerned - I think I'd prefer not to duplicate functionality available on the server, if we can avoid it.